app/services/analysis/gwas/plant_trial_phenotype_csv_builder.rb

```diff
@@ -0,0 +1,66 @@
+class Analysis
+  class Gwas
+    class PlantTrialPhenotypeCsvBuilder
+      def build(plant_trial)
+        documents = exporter(plant_trial).documents
+
+        trait_descriptors = CSV.new(documents.fetch(:trait_descriptors))
+        trait_id_idx = trait_descriptors.readline.index("Trait")
+        trait_ids = trait_descriptors.map { |row| row[trait_id_idx] }
+
+        trait_scoring = CSV.new(documents.fetch(:trait_scoring))
+        trait_scoring_headers = trait_scoring.readline
+        trait_id_indices = trait_ids.map { |trait_id| trait_scoring_headers.index(trait_id) }
+
+        sample_id_idx = trait_scoring_headers.index("Scoring unit name")
+        sample_ids = trait_scoring.map { |row| row[sample_id_idx] }
+
+        # Reset position to the first data line
+        trait_scoring.rewind; trait_scoring.readline
+
+        Result.new(trait_ids, trait_id_indices, sample_ids, trait_scoring)
+      end
+
+      def build_csv(plant_trial)
+        data = build(plant_trial)
+        headers = %w(ID) + data.trait_ids.map { |trait_id| trait_id.gsub(/\s+/, ".") }
+
+        Tempfile.new(["plant-trial-phenotype", ".csv"]).tap do |csv_file|
+          csv_file.write(headers.join(",") + "\n")
```

Comment:
Note to self: use `CSV ` module as it appears that trait and sample names may include unsafe characters.